sdcm/nemesis.py

```diff
@@ -380,6 +380,12 @@ def publish_event(self, disrupt, status=True, data=None):
         severity = Severity.NORMAL if status else Severity.ERROR
         DisruptionEvent(nemesis_name=disrupt, severity=severity, **data).publish()
 
+    def switch_target_node(self, node: BaseNode):
+        self.unset_current_running_nemesis(self.target_node)
+        self.set_current_running_nemesis(node)
+        with NEMESIS_TARGET_SELECTION_LOCK:
+            self.target_node = node
```

Comment:
I think holding the lock once is a bit better, i.e. leave no chance to others to pick something else
```suggestion
    def switch_target_node(self, node: BaseNode):
         with NEMESIS_TARGET_SELECTION_LOCK:
              self.target_node = None
              node.running_nemesis = self.current_disruption
              self.target_node = node
```

Reply:
i think that it is not needed at all - `self.target_node = node` should not be race'y as is only changed by current nemesis (parallel nemesis does not change it as is different object).
and node. and better keep use `set_current_running_nemesis` as this is a subject of refactor and we might want to set things differently.